src/components/algorithm/Algorithms.js

```diff
@@ -152,6 +153,25 @@ class Algorithms extends Component {
 
     const rows = algorithms.map((algorithm) => {
       const { id, name, description, author, language } = algorithm;
+      const quickActions = [
+        <Tooltip title={t('Edit Algorithm')} key="edit">
+          <IconButton aria-label="edit" onClick={() => this.handleEdit(id)}>
+            <EditIcon />
+          </IconButton>
+        </Tooltip>,
+      ];
+      if (author === AUTHOR_USER) {
+        quickActions.push(
+          <Tooltip title={t('Delete algorithm')} key="delete">
+            <IconButton
+              aria-label="delete"
+              onClick={() => this.handleDelete({ id, name })}
+            >
+              <DeleteIcon />
+            </IconButton>
+          </Tooltip>,
```

Comment:
IMO for visual alignment, it would be better to keep the DeleteIcon there for all algos, but have it disabled for non-author algos. that way the edit icons also line up.

Reply:
oh also this might make sense since we're already using this 'design pattern' in the app (i.e. showing disabled icons). so we stay consistent with that approach